ext/standard/crc32.c

```diff
@@ -107,9 +95,53 @@ PHP_FUNCTION(crc32)
 	p += nr_simd;
 #endif
 
+	/* The trailing part */
 	for (; nr--; ++p) {
 		crc = ((crc >> 8) & 0x00FFFFFF) ^ crc32tab[(crc ^ (*p)) & 0xFF ];
 	}
+
+	return crc;
+}
+
+uint32_t crc32_stream_bulk_update(uint32_t crc, php_stream *fp, size_t nr)
+{
+	size_t handled = 0;
+	char buf[1024];
+
+	while (handled < nr) {
+		size_t remaining = nr - handled;
+		size_t n = php_stream_read(fp, buf, (remaining < sizeof(buf)) ? remaining : sizeof(buf));
+		if (n > 0) {
+			crc = crc32_bulk_update(crc, buf, n);
+			handled += n;
+		} else { /* EOF */
+			while (handled < nr) { /* Feed with EOF in case required more than actual stream */
+				CRC32(crc, EOF);
```

Comment:
Could you explain, why this is needed? Also please bear in mind, that `EOF` value might be not portable.

Thanks

Reply:
@jianxind ping, any word on the above?

Thanks

Reply:
> @jianxind ping, any word on the above?
> 
> Thanks

Sorry, I'm having some vacation days recently. I remove both the padding and EOF check in the latest version. Pls check again.

Reply:
Thanks, yeah, i just didn't want to keep it long especially as it took too much time already since your first ping :) See now what was your mention of the `EOF` in [php_stream_getc](https://github.com/php/php-src/blob/master/main/streams/streams.c#L813). `EOF` would be usually `-1` but it doesn't seem portable and i also don't find any mention of padding in `CRC32` papers :/ I'd still suggest to keep your more clean latest, otherwise could revert to previous if there are BC issues. Great vacation otherwise, if you're still on it, this is sure not urgent.

Thanks!